app/controllers/declaration/SummaryController.scala

```diff
@@ -65,16 +67,21 @@ class SummaryController @Inject() (
   }
 
   private def amendmentSummaryPage()(implicit request: JourneyRequest[_]): Future[Result] =
-    getValue(submissionUuid).fold {
-      errorHandler.internalError("Session on /saved-summary (for draft amendment) does not include 'submissionUuid'??")
-    } { submissionId =>
-      val html = Html(
-        amendment_summary(submissionId)
-          .toString()
-          .replace(s"?$lastUrlPlaceholder", "")
-      )
-      Future.successful(Ok(html))
-    }
+    request.cacheModel.declarationMeta.parentDeclarationId map { parentDeclarationId =>
+      connector.findSubmissionByLatestDecId(parentDeclarationId) flatMap { submissionIdOpt =>
+        submissionIdOpt map { submission =>
+          Future.successful(
+            Ok(
+              Html(
+                amendment_summary()
+                  .toString()
+                  .replace(s"?$lastUrlPlaceholder", "")
+              )
+            ).addingToSession(submissionUuid -> submission.uuid)
+          )
+        } getOrElse errorHandler.internalError("Cannot associate submission with parentDecId")
+      }
+    } getOrElse errorHandler.internalError("ParentDecId cannot be found")
```

Comment:
I would add to the 2 error messages the declaration's id, as it would make easier to locate the affected declaration.